tests/s3.rs

```diff
@@ -1,96 +1,165 @@
 #![cfg(feature = "s3")]
```

Comment:
These integration tests were a total rewrite, no need to compare to the old on the left, it's all new.